src/main/java/software/amazon/smithy/lsp/project/ProjectDependencyResolver.java

```diff
@@ -0,0 +1,117 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import software.amazon.smithy.build.SmithyBuild;
+import software.amazon.smithy.build.model.MavenConfig;
+import software.amazon.smithy.build.model.MavenRepository;
+import software.amazon.smithy.cli.EnvironmentVariable;
+import software.amazon.smithy.cli.dependencies.DependencyResolver;
+import software.amazon.smithy.cli.dependencies.MavenDependencyResolver;
+import software.amazon.smithy.cli.dependencies.ResolvedArtifact;
+import software.amazon.smithy.lsp.util.Result;
+
+/**
+ * Resolves all Maven dependencies and {@link ProjectDependency} for a project.
+ *
+ * <p>Resolving a {@link ProjectDependency} is as simple as getting its path
+ * relative to the project root, but is done here in order to be loaded the
+ * same way as Maven dependencies.
+ */
+final class ProjectDependencyResolver {
+    // Taken from smithy-cli ConfigurationUtils
+    private static final Supplier<MavenRepository> CENTRAL = () -> MavenRepository.builder()
+            .url("https://repo.maven.apache.org/maven2")
+            .build();
+
+    private ProjectDependencyResolver() {
+    }
+
+    static Result<List<Path>, Exception> resolveDependencies(Path root, ProjectConfig config) {
+        return Result.ofFallible(() -> {
+            List<Path> deps = ProjectDependencyResolver.create(config).resolve()
+                    .stream()
+                    .map(ResolvedArtifact::getPath)
+                    .collect(Collectors.toCollection(ArrayList::new));
+            config.getDependencies().forEach((projectDependency) -> {
+                // TODO: Not sure if this needs to check for existence
+                Path path = root.resolve(projectDependency.getPath()).normalize();
+                deps.add(path);
+            });
+            return deps;
+        });
+    }
+
+    // Taken (roughly) from smithy-cli ClasspathAction::resolveDependencies
+    private static DependencyResolver create(ProjectConfig config) {
+        // DependencyResolver delegate = new MavenDependencyResolver(EnvironmentVariable.SMITHY_MAVEN_CACHE.get());
+        // long lastModified = config.getLastModifiedInMillis();
+        // DependencyResolver resolver = new FileCacheResolver(getCacheFile(config), lastModified, delegate);
+
+        // TODO: Seeing what happens if we just don't use the file cache. When we do, at least for testing, the
+        //  server writes a classpath.json to build/smithy/ which is used by all tests, messing everything up.
+        DependencyResolver resolver = new MavenDependencyResolver(EnvironmentVariable.SMITHY_MAVEN_CACHE.get());
+
+        Set<MavenRepository> configuredRepositories = getConfiguredMavenRepos(config);
+        configuredRepositories.forEach(resolver::addRepository);
+
+        // TODO: Support lock file ?
```

Comment:
Not sure why the `?`, but yea we will need to update the CLI's APIs to make this possible

Reply:
Maybe the `?` was because there might be an argument to be made for not using the lockfile for the language server. Every time you update a dep in smithy-build.json, the language server would have to be like, hey, run smithy lock. You'd get an error from the CLI anyways if you tried to build. Plus the language server then has to track changes to smithy-build.json (so you can get language features), but ignore them if the lockfile is out of sync.

Just some initial thoughts thinking out loud. The direction I've been going in is to have the language server just pay attention to 'source' files, so its always up to date on the latest changes you've made.

Reply:
We can circle back to this if it becomes an issue.